wgsl/index.bs

```diff
@@ -2602,51 +2605,109 @@ programmable parts of pipelines:
 * <dfn noexport>vertex</dfn>
 * <dfn noexport>fragment</dfn>
 
-<pre class='def'>
-pipeline_stage
-  : COMPUTE
-  | VERTEX
-  | FRAGMENT
-</pre>
-
 Each shader stage has its own set of features and constraints, described elsewhere.
 
-## Entry point declaration TODO ## {#entry-point-decl}
+## Entry point declaration ## {#entry-point-decl}
 
-The `entry_point` declares an entry point into the module. The entry points may
-be forward declarations but the functions referenced must be declared in the
-file.
+An <dfn noexport>entry point</dfn> is a function that is invoked to perform
+the work for a particular [=shader stage=].
 
-The input and output parameters to the entry point are determined by which
-global variables are used in the function and any called functions.
+Specify a `stage` attribute on a function declaration to declare that function
+as an entry point.
 
-The entry point function must be declared with no arguments and returning `void`.
+When configuring the stage in the pipeline, the entry point is specified by providing
+the [SHORTNAME] module and the entry point's function name.
 
-<pre class='def'>
-entry_point_decl
-  : ENTRY_POINT pipeline_stage EQUAL IDENT
-  | ENTRY_POINT pipeline_stage AS STRING_LITERAL EQUAL IDENT
-  | ENTRY_POINT pipeline_stage AS IDENT EQUAL IDENT
-</pre>
+An entry point function must have no parameters,
+and its return type must be [=void=].
 
 <div class='example' heading='Entry Point'>
   <xmp>
-    entry_point vertex = main
-       OpEntryPoint Vertex %vtx_main "vtx_main" %gl_FragColor
+    [[stage(vertex)]]
+    fn vtx_main() -> void { gl_Position = vec4<f32>(); }
+       # OpEntryPoint Vertex %vtx_main "vtx_main" %gl_Position
+
+    [[stage(fragment)]]
+    fn frag_main -> void { gl_FragColor = vec4<f32>(); }
+       # OpEntryPoint Fragment %frag_main "frag_main" %gl_FragColor
 
-    entry_point fragment as “frag_main” = main
-       OpEntryPoint Fragment %main "frag_main" %gl_FragColor
+    [[stage(compute)]]
+    fn main() -> void { }
+       # OpEntryPoint GLCompute %main "main"
+  </xmp>
+</div>
+
+The set of <dfn noexport>functions in a shader stage</dfn> is the union of:
+
+* The entry point function for the stage.
+* The targets of function calls from within the body of a function
+    in the shader stage, whether or not that call is executed.
```

Comment:
it would be useful to define "statically reachable" somewhere and refer to it in places like this instead of redefining